core-web/libs/sdk/experiments/src/lib/sdk-experiments.ts

```diff
@@ -1,14 +1,117 @@
+import { API_EXPERIMENTS_URL } from './contants';
+import { IsUserIncludedApiResponse, SdkExperimentConfig } from './models';
+import { persistData } from './persistence/persistence';
+import { Logger } from './utils/utils';
+
 /**
- * Represents an SDK Experiment.
- * @class
+ * This class handles all operations related to fetching, storing, parsing, and navigating data for SdkExperiments.
+ * Utilizes the Jitsu SDK to send events.
+ *
+ * The operations are
+ * 1. Fetch and store data ✅
+ * 2. Parse data for backend use
+ * 3. Detect and redirect to the appropriate variant
+ * 4. Send event using Jitsu SDK
  */
-export class DotSdkExperiment {
+export class SdkExperiments {
+    private static instance: SdkExperiments;
+
+    constructor(private config: SdkExperimentConfig) {
+        // api-key could be an empty string but server should not be empty or undefined.
+        if (!this.config.server) {
+            throw new Error('server must be provided and should not be empty!');
+        }
+
+        Logger(
+            `SDK instanced with ${JSON.stringify(config)} configuration`,
+            this.getIsDebugActive()
+        );
+    }
+
+    /**
+     * Retrieves or initializes the instance of SdkExperiments class.
+     * If the instance does not exist, it creates a new instance with the provided configuration and calls the 'getExperimentData' method.
+     *
+     * @param {SdkExperimentConfig} config - The configuration object for initializing the SdkExperiments instance.
+     * @return {SdkExperiments} - The instance of the SdkExperiments class.
+     */
+    public static getInstance(config: SdkExperimentConfig): SdkExperiments {
+        if (!SdkExperiments.instance) {
+            SdkExperiments.instance = new SdkExperiments(config);
+            this.instance.getExperimentData();
+        }
+
+        return SdkExperiments.instance;
+    }
+
+    /**
+     * Fetches experiments from the server.
+     *
+     * @private
+     * @returns {Promise<Object>} - A promise that resolves to the entity object if the request is successful.
+     * @throws {Error} - If there is an HTTP error.
+     */
+    private async fetchExperiments(): Promise<IsUserIncludedApiResponse['entity']> {
+        const response: Response = await fetch(`${this.config.server}${API_EXPERIMENTS_URL}`, {
+            method: 'POST',
+            headers: {
+                Accept: 'application/json',
+                'Content-Type': 'application/json'
+            }
+        });
```

Comment:
The endpoint to get the Running Experiment is a `POST`, because right now use a body to send a excluded id of arrays but that will be deleted. Maybe I can talk with BE to change from `POST` to `GET`. @freddyDOTCMS 
<img width="1042" alt="Screenshot 2024-02-24 at 2 13 28 PM" src="https://github.com/dotCMS/core/assets/1909643/f9cf0698-91e5-4320-ac1e-9dcdf37eb1e3">